--- vendor/bundle/jruby/1.9/gems/logstash-input-s3-2.0.2/lib/logstash/inputs/s3.rb
+++ vendor/bundle/jruby/1.9/gems/logstash-input-s3-2.0.2/lib/logstash/inputs/s3.rb
@@ -65,7 +65,7 @@ class LogStash::Inputs::S3 < LogStash::Inputs::Base
 
   # Set the directory where logstash will store the tmp files before processing them.
   # default to the current OS temporary directory in linux /tmp/logstash
-  config :temporary_directory, :validate => :string, :default => File.join(Dir.tmpdir, "logstash")
+  config :temporary_directory, :validate => :string, :default => "/var/lib/logstash/S3_input_temp"
 
   public
   def register
