--- vendor/bundle/jruby/1.9/gems/logstash-input-s3-2.0.2/lib/logstash/inputs/s3.rb
+++ vendor/bundle/jruby/1.9/gems/logstash-input-s3-2.0.2/lib/logstash/inputs/s3.rb
@@ -67,6 +67,10 @@ class LogStash::Inputs::S3 < LogStash::Inputs::Base
   # default to the current OS temporary directory in linux /tmp/logstash
   config :temporary_directory, :validate => :string, :default => "/var/lib/logstash/S3_input_temp"
 
+  # Set the username that will be used for server starting#
+  # https://github.com/vitvegl/elk-deploy/blob/qg/SOURCES/logstash.service
+  config :temporary_directory_owner, :validate => :string, :default => "logstash"
+
   public
   def register
     require "fileutils"
@@ -88,11 +92,13 @@ class LogStash::Inputs::S3 < LogStash::Inputs::Base
       end
     end
 
-    unless @backup_to_dir.nil?
-      Dir.mkdir(@backup_to_dir, 0700) unless File.exists?(@backup_to_dir)
-    end
+    unless Process.euid == 0
+      unless @backup_to_dir.nil?
+        Dir.mkdir(@backup_to_dir, 0700) unless File.exists?(@backup_to_dir)
+      end
 
-    FileUtils.mkdir_p(@temporary_directory) unless Dir.exist?(@temporary_directory)
+      FileUtils.mkdir_p(@temporary_directory) unless Dir.exist?(@temporary_directory)
+    end
   end # def register
 
   public
